libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -474,6 +477,19 @@ void AP_Baro::init(void)
 #endif
     }
     
+#if HAL_WITH_UAVCAN
+    // If there is place left - allocate one UAVCAN based baro
+    if ((AP_BoardConfig::get_can_enable() != 0) && (hal.can_mgr != nullptr))
+    {
+        if(_num_drivers < BARO_MAX_DRIVERS && _num_sensors < BARO_MAX_INSTANCES)
+        {
+            printf("Creating AP_Baro_UAVCAN\n\r");
+            drivers[_num_drivers] = new AP_Baro_UAVCAN(*this);
```

Comment:
We don't have "type" parameter for these device to specifically allow or disable it. However at time of boot, we might not know if it is present, we will find out much later.